### ISSUE TITLE
Support for python 3.12

### ISSUE THREAD

**Issue by Sam1320** [modal-labs/modal-client#1056]
I tried to use modal with the newest python version (3.12) but I kept getting this error:

```
➜  modal
Traceback (most recent call last):
  File "/opt/homebrew/Caskroom/miniconda/base/envs/llms/bin/modal", line 5, in <module>
    from modal.__main__ import main
  File "/opt/homebrew/Caskroom/miniconda/base/envs/llms/lib/python3.12/site-packages/modal/__init__.py", line 5, in <module>
    from .app import container_app, is_local
  File "/opt/homebrew/Caskroom/miniconda/base/envs/llms/lib/python3.12/site-packages/modal/app.py", line 17, in <module>
    from .object import _Object
  File "/opt/homebrew/Caskroom/miniconda/base/envs/llms/lib/python3.12/site-packages/modal/object.py", line 23, in <module>
    _BLOCKING_O = synchronize_api(O)
                  ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/llms/lib/python3.12/site-packages/modal_utils/async_utils.py", line 31, in synchronize_api
    return synchronizer.create_blocking(obj, blocking_name, target_module=target_module)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/llms/lib/python3.12/site-packages/synchronicity/synchronizer.py", line 687, in create_blocking
    wrapped = self._wrap(obj, Interface.BLOCKING, name, target_module=target_module)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/llms/lib/python3.12/site-packages/synchronicity/synchronizer.py", line 623, in _wrap
    if self._wrapped_attr not in obj.__dict__:

```
    
  When I downgraded python to `3.11.5` it worked fine!.

**Comment by priamai** [comment]
The last supported version I believe is **3.11.6** correct?

**Comment by erikbern** [comment]
It should work with all versions of 3.11!

**Comment by pdurusau** [comment]
I installed Python 3.11 and created a venv but it still failed with the Python 3.12 message. 
Uninstalling the MS Python app still left the local package string with Python 3.12 and so it still failed. 
Is Modal testing the environmental variables for Python 3.12 and not Python 3.12 artifacts in the code? (the code I was trying was as basic as it gets) 
Runs after I deleted the environmental string with Python 3.12. Just seems like an odd check.
